tests/Start-DbaAgentJob.Tests.ps1

```diff
@@ -1,42 +1,47 @@
-$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
-Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
-. "$PSScriptRoot\constants.ps1"
-
-Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
-    Context "Validate parameters" {
-        [object[]]$params = (Get-Command $CommandName).Parameters.Keys | Where-Object {$_ -notin ('whatif', 'confirm')}
-        [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Job', 'ExcludeJob', 'InputObject', 'AllJobs', 'Wait', 'WaitPeriod', 'SleepPeriod', 'EnableException'
-        $knownParameters += [System.Management.Automation.PSCmdlet]::CommonParameters
-        It "Should only contain our specific parameters" {
-            (@(Compare-Object -ReferenceObject ($knownParameters | Where-Object {$_}) -DifferenceObject $params).Count ) | Should Be 0
-        }
-    }
-}
-
-Describe "$CommandName Integration Tests" -Tag "IntegrationTests" {
-    Context "Start a job" {
-        BeforeAll {
-            $jobName = "dbatoolsci_job_$(get-random)"
-            $null = New-DbaAgentJob -SqlInstance $script:instance2,$script:instance3 -Job $jobName
-            $null = New-DbaAgentJobStep -SqlInstance $script:instance2,$script:instance3 -Job $jobName -StepName dbatoolsci_jobstep1 -Subsystem TransactSql -Command 'select 1'
-        }
-        AfterAll {
-            $null = Remove-DbaAgentJob -SqlInstance $script:instance2,$script:instance3 -Job $jobName -Confirm:$false
-        }
-
-        It "returns a CurrentRunStatus of not Idle and supports pipe" {
-            $results = Get-DbaAgentJob -SqlInstance $script:instance2 -Job $jobName | Start-DbaAgentJob
-            $results.CurrentRunStatus -ne 'Idle' | Should Be $true
-        }
-
-        It "does not run all jobs" {
-            $null = Start-DbaAgentJob -SqlInstance $script:instance2 -WarningAction SilentlyContinue -WarningVariable warn
-            $warn -match 'use one of the job' | Should Be $true
-        }
-
-        It "returns on multiple server inputs" {
-            $results2 = Start-DbaAgentJob -SqlInstance $script:instance2, $script:instance3 -Job $jobName
-            ($results2.SqlInstance).Count | Should -Be 2
-        }
-    }
+$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
+. "$PSScriptRoot\constants.ps1"
+
+Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
+    Context "Validate parameters" {
+        [array]$params = ([Management.Automation.CommandMetaData]$ExecutionContext.SessionState.InvokeCommand.GetCommand($CommandName, 'Function')).Parameters.Keys
+        [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Job', 'ExcludeJob', 'InputObject', 'AllJobs', 'Wait', 'Parallel', 'WaitPeriod', 'SleepPeriod', 'EnableException'
+        $knownParameters += [System.Management.Automation.PSCmdlet]::CommonParameters

```

Comment:
This line needs to be removed.

Reply:
The full context block should just be:

```powershell
    Context "Validate parameters" {
        [array]$params = ([Management.Automation.CommandMetaData]$ExecutionContext.SessionState.InvokeCommand.GetCommand($CommandName, 'Function')).Parameters.Keys
        [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Job', 'ExcludeJob', 'InputObject', 'AllJobs', 'Wait', 'Parallel', 'WaitPeriod', 'SleepPeriod', 'EnableException'

        It "Should only contain our specific parameters" {
            Compare-Object -ReferenceObject $knownParameters -DifferenceObject $params | Should -BeNullOrEmpty
        }
    }
}
```

Reply:
The two things this does is excluding needing to deal with the common parameters. Then also we are pulling the parameters from cache instead (since our prep script imports the module) of calling `Get-Command`, which shaves off a good few seconds of time on the test run.

Reply:
Making the change and committing in a few... thanks for the info.

Reply:
Before I commit, I'm going to wait for a response from @potatoqualitee since the code she asked me to include in the test is failing as well.  In addition, the server test failed too - and I did not change that.  That was added from previous change.

Reply:
We need to get this PR finished so I've fixed the tests.

The two items are failing because you will never get null or not idle while the job runs in one millisecond. I changed the command for the job step to just sit and do a wait for 5 minutes. This will ensure we get the status the test are looking for to pass. The time of the job step won't affect the test itself since we drop it at the end.

Reply:
@wsmelton I thought I had suggested that somewhere but cannot find the comment anymore